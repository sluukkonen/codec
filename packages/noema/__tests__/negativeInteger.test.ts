import { negativeInteger } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("negativeInteger", () => {
  test("should accept negative integers", () => {
    expectParseSuccess(negativeInteger, -1)
    expectParseSuccess(negativeInteger, -2)
    expectParseSuccess(negativeInteger, -3)
  })

  test("should not accept 0", () => {
    expectParseFailure(negativeInteger, 0)
  })

  test("should not accept positive integers", () => {
    expectParseFailure(negativeInteger, 1)
    expectParseFailure(negativeInteger, 2)
    expectParseFailure(negativeInteger, 2)
  })

  test("should not accept non-integers", () => {
    expectParseFailure(negativeInteger, -0.5)
    expectParseFailure(negativeInteger, 0.5)
  })

  test("should not accept NaN", () => {
    expectParseFailure(negativeInteger, NaN)
  })
})
