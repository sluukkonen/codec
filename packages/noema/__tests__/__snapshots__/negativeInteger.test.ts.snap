// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`negativeInteger should not accept 0 1`] = `
[
  {
    "code": "too_large",
    "max": -1,
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept NaN 1`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept non-integers 1`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept non-integers 2`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept positive integers 1`] = `
[
  {
    "code": "too_large",
    "max": -1,
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept positive integers 2`] = `
[
  {
    "code": "too_large",
    "max": -1,
    "path": [],
  },
]
`;

exports[`negativeInteger should not accept positive integers 3`] = `
[
  {
    "code": "too_large",
    "max": -1,
    "path": [],
  },
]
`;
